### PR TITLE
Updated the git pull-request command to use personal access token instead of basic auth.

### LINKED ISSUE #863: git pull-request - Github API decprication notice
Could not find this in issues, pls close if duplicate. 

on using `git pull-request` I received the following content as email from github. 

can this feature have personal access tokens or ssh based authentication? 

```
Basic authentication using a password to the API is deprecated and will soon no longer work. Visit https://developer.github.com/changes/2020-02-14-deprecating-password-auth/ for more information around suggested workarounds and removal dates.

Thanks,
The GitHub Team
```

### PR DESCRIPTION
**Pull request by v-y-a-s**
Updated the `git pull-request` command to use personal access token instead of basic auth.

### CODE REVIEW

**Comment by v-y-a-s** [comment]
#[#863 ](url)

File: man/git-pull-request.md [hunk 1]
```diff
@@ -11,6 +11,14 @@ Create pull request for a project on GitHub via command line.
 
 Uses the email from `git config user.email` to open the pull request.
 
+Personal access token is required for making API call to open the pull request. 
+
+Makre sure the personal access token has the right `OAuth` scopes for the repo(s)
```
Comment (spacewander) [inline man/git-pull-request.md:16]: 'Make sure'

File: bin/git-pull-request [hunk 2]
```diff
@@ -38,6 +38,13 @@ fi
 
 test -z "$user" && abort "git config user.email required"
```
Comment (spacewander) [inline bin/git-pull-request:39]: It seems that we can remove `$user` and its doc now?

File: man/git-pull-request.md [hunk 3]
```diff
@@ -11,6 +11,14 @@ Create pull request for a project on GitHub via command line.
...
+
+use `git config --global --add git-extras.personal-access-token <your-personal-access-token>`
```
Comment (spacewander) [inline man/git-pull-request.md:18]: Sentence should be started with capital letter.

File: man/git-pull-request.md [hunk 4]
```diff
@@ -11,6 +11,14 @@ Create pull request for a project on GitHub via command line.
...
+
+if using mutiple accounts, override the global value in the specific repo using `git config git-extras.personal-access-token <other-acc-personal-access-token>`
```
Comment (spacewander) [inline man/git-pull-request.md:20]: Ditto

File: bin/git-pull-request [hunk 5]
```diff
@@ -38,6 +38,13 @@ fi
...
 
+# personal access token
+# config name is personal-access-token '_' is not allowed in git config
+
+personal_access_token=$(git config git-extras.personal-access-token)
```
Comment (spacewander) [inline bin/git-pull-request:44]: Please add `github` in the config name, since the token is only for GitHub.

File: man/git-pull-request.md [hunk 6]
```diff
@@ -9,7 +9,13 @@ git-pull-request(1) -- Create pull request for GitHub project
 
 Create pull request for a project on GitHub via command line.
 
-Uses the email from `git config user.email` to open the pull request.
+Personal access token is required for making API call to open the pull request in Github.
```
Comment (spacewander) [inline man/git-pull-request.md:12]: s/Github/GitHub

Comment (v-y-a-s) [reply man/git-pull-request.md:12]: Sorry I did not get this

**Comment by spacewander** [comment]
@v-y-a-s 
Merged. Thanks!

**Pull request merged**